spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/catalog/FunctionAroundWrapper.java

```diff
@@ -27,33 +27,28 @@
  * If registered as bean it will be autowired into {@link FunctionInvocationWrapper}.
  * Keep in mind that it only affects imperative invocations where input is {@link Message}
  *
- * NOTE: This API is experimental and and could change without notice. It is
+ * NOTE: This API is experimental and could change without notice. It is
  * intended for internal use only (e.g., spring-cloud-sleuth)
  *
  * @author Oleg Zhurakousky
+ * @author Artem Bilan
  * @since 3.1
  */
 public abstract class FunctionAroundWrapper implements BiFunction<Object, FunctionInvocationWrapper, Object> {
 
 	@Override
 	public final Object apply(Object input, FunctionInvocationWrapper targetFunction) {
 		String functionalTracingEnabledStr = System.getProperty("spring.sleuth.function.enabled");
```

Comment:
I am guessing this is using system property rather than a Spring property source as functions are useable outside of envs that provide the sys props as spring properties (like Spring Boot does)?

Reply:
Yeah, and the above code is really a fail safe and somewhat redundant as it is used as conditional in sleuth. 
That said, in case thing unexpectedly change in sleuth it provides ability to disconnect sleuth at the function level.
Once the sleuth functionality related to tracing wrapper becomes stable, we can actually remove that all together.

Reply:
Will simplify as Chris suggests.
And won't look into `Environment` as it has to be in Spring Boot since the property will go away as Oleg explains.